lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -2480,31 +2906,260 @@ namespace Js
+    {
+        return IsObjTypeSpecEquivalentHelper(type, attributes, entry);
+    }
+#endif
+
+    DescriptorFlags SimplePathTypeHandlerWithAttr::GetSetter(DynamicObject* instance, PropertyId propertyId, Var* setterValue, PropertyValueInfo* info, ScriptContext* requestContext)
+    {
+        PropertyIndex propertyIndex = GetTypePath()->LookupInline(propertyId, GetPathLength());
+        if (propertyIndex == Constants::NoSlot)
+        {
+            return __super::GetSetter(instance, propertyId, setterValue, info, requestContext);
+        }
+        ObjectSlotAttributes attr = attributes[propertyIndex];
+        if (attr & ObjectSlotAttr_Deleted)
+        {
+            return None;
+        }
+
+        Assert(!(attr & ObjectSlotAttr_Accessor));
+
+        if (attr & ObjectSlotAttr_Writable)
+        {
+            return WritableData;
+        }
+        return Data;
+    }
+
+    DescriptorFlags SimplePathTypeHandlerWithAttr::GetSetter(DynamicObject* instance, JavascriptString* propertyNameString, Var* setterValue, PropertyValueInfo* info, ScriptContext* requestContext)
+    {
+        PropertyRecord const* propertyRecord;
+        instance->GetScriptContext()->GetOrAddPropertyRecord(propertyNameString, &propertyRecord);
+        return this->GetSetter(instance, propertyRecord->GetPropertyId(), setterValue, info, requestContext);
+    }
+
+    BOOL SimplePathTypeHandlerWithAttr::GetProperty(DynamicObject* instance, Var originalInstance, PropertyId propertyId, Var* value, PropertyValueInfo* info, ScriptContext* requestContext)
+    {
+        PropertyIndex index = GetTypePath()->LookupInline(propertyId, GetPathLength());
+        if (index == Constants::NoSlot)
+        {
+            return __super::GetProperty(instance, originalInstance, propertyId, value, info, requestContext);
+        }
+
+        *value = instance->GetSlot(index);
+        PropertyValueInfo::Set(info, instance, index, ObjectSlotAttributesToPropertyAttributes(attributes[index]));
+#ifdef SUPPORT_FIXED_FIELDS_ON_PATH_TYPES
+        if (FixPropsOnPathTypes() && (index >= this->GetTypePath()->GetMaxInitializedLength() || this->GetTypePath()->GetIsFixedFieldAt(index, GetPathLength())))
+        {
+            PropertyValueInfo::DisableStoreFieldCache(info);
+        }
+#endif
+        return true;
+    }
+
+    BOOL SimplePathTypeHandlerWithAttr::GetProperty(DynamicObject* instance, Var originalInstance, JavascriptString* propertyNameString, Var* value, PropertyValueInfo* info, ScriptContext* requestContext)
+    {
+        // Consider: Implement actual string hash lookup
+        Assert(requestContext);
+        PropertyRecord const* propertyRecord;
+        char16 const * propertyName = propertyNameString->GetString();
+        charcount_t const propertyNameLength = propertyNameString->GetLength();
+
+        if (instance->HasObjectArray())
+        {
+            requestContext->GetOrAddPropertyRecord(propertyName, propertyNameLength, &propertyRecord);
```

Comment:
use `propertyNameString` directly? #3875 adds `PropertyString` and `LiteralStringWithPropertyStringPtr` check.

Reply:
Thanks. This is actually a copy of some very old logic. Should we change it everywhere?